src/AbstractSQLCompiler.ts

```diff
@@ -234,13 +234,14 @@ export interface AbstractSqlField {
 	fieldName: string;
 	dataType: string;
 	required?: boolean;
-	index: string;
+	index?: string;
```

Comment:
Any clue on why this became nullable?

Reply:
It was always nullable and this was just typed incorrectly

Reply:
As to why I changed it, when creating a computed field you need to be able to omit it and that creation is done in typescript so the types had to be correct (the place it's done for sbvr is https://github.com/balena-io-modules/lf-to-abstract-sql/blob/v3.0.1/lf-to-abstract-sql.ometajs which doesn't support typescript)